src/registry/domain/repository.js

```diff
@@ -175,6 +175,19 @@ module.exports = function(conf){
     getStaticFilePath: function(componentName, componentVersion, filePath){
       return this.getComponentPath(componentName, componentVersion) + (conf.local ? settings.registry.localStaticRedirectorPath : '') + filePath;
     },
+    getTemplates: function(){
+      if (conf.templates) {
+        return conf.templates.map(function(template){
+          var info = require(template).getInfo();
+          return {
+            type: info.type,
+            version: info.version,
+            externals: info.externals
+          };
+        });
+      }
+      return [];
+    },
```

Comment:
This is good, but I am thinking performance here. This if+map+getInfo() is executed on each component execution (when the context is baked) by making this call and I don't think there is actually any need for that, as the templates are kind of statically initialised during registry's start anyways. What about moving this to the constructor on the top (around line 17) and memorising that in a variable, so that this method just exposes it?